src/datasources/transaction-api/transaction-api.service.ts

```diff
@@ -144,7 +144,7 @@ export class TransactionApi implements ITransactionApi {
         contractAddress,
       );
       const url = `${this.baseUrl}/api/v1/contracts/${contractAddress}`;
-      return await this.dataSource.get(cacheDir, url);
+      return await this.dataSource.getAndCacheError(cacheDir, url);
```

Comment:
I think the added methods add complexity to the relatively simple API that we currently have. I think we should clarify if we want to cache 404s and if we want to cache 404s would it be for every single endpoint?

Reply:
I unified the condition of both functions in [bf97b75](https://github.com/safe-global/safe-client-gateway-nest/pull/295/commits/bf97b75b8dc54807875a63f8b17f9c3884f9ce12)

I agree, it doesn't make sense to have different conditions here. I chose `cached != null` because this covers the case the cached data is both `null` or `undefined`, but it discards the case the cached data is any other _falsy_ value.